### ISSUE TITLE
Template variables in aggregators

### ISSUE THREAD

**Issue by fbeneventi** [grafana/kairosdb-datasource#67]
Hi,

I need to use a template variable in an aggregator expression as for example:

$N = 24

and then in aggregator I try to do:

scale($N)

where $N is a "custom" type template variable.

Eexecuting this query I have the following error:

query.metric[1].aggregators[1].factor For input string: "$N"

It seems that variables are not interpolated in aggregators. Is there a way/plan to do this?

thanks!
Francesco

**Comment by jifwin** [comment]
Templating of aggregators parameters is a bit more tricky, but definitely doable. The question is how to handle multi-value variables with many aggregators defined. E.g.

var1=[1,2,3], AVG($var1,HOUR) + MAX($var1, HOUR)
Would you expect 9 series (cartesian) or 3 (1-to-1 matching)?

Or maybe multivalue variables shouldn't be allowed allowed at all?

**Comment by gucharbon** [comment]
I'm not sure I understood your point.
I'm okay with not allowing multi variables for variable in Aggregators.
If we wanted to create several visualizations based on each unit or value of aggregator it would make sense, but this seems far too complicated and not really useful.
We would already gain an awesome feature if we  could use Single variable in aggregators

**Comment by fbeneventi** [comment]
Yes, currently only "custom" variables seem not working in aggregators and for now it should be ok to support just single value variables.

**Comment by gucharbon** [comment]
Oh really ? That means that if I use any other variable, like store my desired Units in a mysql table I can query them and use them in the query builder ? I didn't know that

**Comment by jifwin** [comment]
I don't think so. It doesn't matter what type of variable it is.

**Comment by jifwin** [comment]
@gucharbon Are you OK with support for single-value variables?

**Comment by gucharbon** [comment]
@jifwin  Definitely ! :+1:
